animation_url/src/routes/[slug]/+page.ts

```diff
@@ -31,19 +49,64 @@ export async function load({ url }) {
   });
   const httpUri = ipfsGatewayUri + metadata.image.slice(7);
 
-  // get pilot
-  let pilot;
-  const sessions = await tableland.read(
-    `SELECT end_time FROM ${deployment.pilotSessionsTable} WHERE rig_id = ${rigId} AND end_time is null;`,
-    { output: "objects" }
-  );
-  if (sessions && sessions.length > 0) {
-    pilot = trainer;
-  }
+  const pilot = await getPilot(rigId);
 
   return {
     imageUrl: httpUri,
     badges: [],
     pilot,
   };
 }
+
+const getPilot = async function (rigId, pilotAddress, pilotId): Promise<string> {
```

Comment:
This holds all the display logic for pilots and trainering and parked.  I believe I got everything covered, but definitely worth looking close at this since there are so many conditionals.

Reply:
I think this can be simplified to just do one query right? similar to what i posted in discord?

Reply:
ie, we don't care about any sessions where end_time is NOT null

Reply:
I'll update now...

Reply:
@sanderpick Sorry again about the confusion here.  I updated things.  There's only one query now, and the logic assumes that a given rig can only ever have one row where end_time is null.  Let me know if this looks right.